jax/lax/lax.py

```diff
@@ -4803,7 +4811,23 @@ def _prescan_power_of_two(x, axis: int, op: Callable, unit):
   return x, total
 
 
-def _parallel_prefix_scan(x, axis: int, op: Callable, unit):
+def _parallel_prefix_scan(x, axis: int, op: Callable):
+  if op is max:
+    if onp.issubdtype(x.dtype, onp.integer):
+      unit = onp.iinfo(x.dtype).min
+    else:
+      unit = dtypes.finfo(x.dtype).min
+  elif op is min:
+    if onp.issubdtype(x.dtype, onp.integer):
+      unit = onp.iinfo(x.dtype).max
+    else:
+      unit = dtypes.finfo(x.dtype).max
```

Comment:
If `x` only contains positive infinities, the result should be infinity. To achieve that, I think we need the unit to default to `np.inf` or `-np.inf`, rather than the largest/smallest finite float value.

Reply:
Done.